tests/unit/TestResultsService.unitTest.js

```diff
@@ -628,25 +628,75 @@ describe('setExpiryDateAndCertificateNumber', () => {
+            expectedExpiryDate.setFullYear(new Date().getFullYear() + 1)
+            expectedExpiryDate.setDate(new Date().getDate() - 1)
+            expect((response.testTypes[0].testExpiryDate).split('T')[0]).to.equal(expectedExpiryDate.toISOString().split('T')[0])
+            expect(response.testTypes[0].certificateNumber).to.equal(response.testTypes[0].testNumber)
+            expect(response.testTypes[1].testExpiryDate).to.equal(undefined)
+            expect(response.testTypes[1].certificateNumber).to.equal(undefined)
+            expect(response.testTypes[2].testExpiryDate).to.equal(undefined)
+            expect(response.testTypes[2].certificateNumber).to.equal(undefined)
+          })
+      })
+    })
+    context('for hgv and trl vehicle types', () => {
+      context('when there is no certificate issued for this vehicle', () => {
+        it('should set the expiry date to last day of current month + 1 year', () => {
+          testResultsDAOMock.testResultsResponseMock = []
+          testResultsDAOMock.numberOfrecords = 0
+          testResultsDAOMock.numberOfScannedRecords = 0
+          const testResultsService = new TestResultsService(testResultsDAOMock)
+          let mockData = testResultsMockDB[15]
+          return testResultsService.setExpiryDateAndCertificateNumber(mockData)
+            .then(response => {
+              const expectedExpiryDate = dateFns.addYears(dateFns.lastDayOfMonth(new Date()), 1)
+              expect((response.testTypes[0].testExpiryDate).split('T')[0]).to.equal(expectedExpiryDate.toISOString().split('T')[0])
+            })
+        })
+      })
+      context('when there is a certificate issued for this vehicle that expired', () => {
+        it('should set the expiry date to last day of current month + 1 year', () => {
+          testResultsDAOMock.testResultsResponseMock = Array.of(testResultsMockDB[15])
+          testResultsDAOMock.numberOfrecords = 1
+          testResultsDAOMock.numberOfScannedRecords = 1
+          const pastExpiryDate = dateFns.subMonths(new Date(), 1)
+          testResultsDAOMock.testResultsResponseMock[0].testTypes[0].testExpiryDate = pastExpiryDate
+          const testResultsService = new TestResultsService(testResultsDAOMock)
+          let mockData = testResultsMockDB[15]
+          return testResultsService.setExpiryDateAndCertificateNumber(mockData)
+            .then(response => {
+              const expectedExpiryDate = dateFns.addYears(dateFns.lastDayOfMonth(new Date()), 1)
+              expect((response.testTypes[0].testExpiryDate).split('T')[0]).to.equal(expectedExpiryDate.toISOString().split('T')[0])
+            })
+        })
+      })
+      context('when there is a certificate issued for this vehicle that is expiring more than two months in the future', () => {
+        it('should set the expiry date to last day of current month + 1 year', () => {
+          testResultsDAOMock.testResultsResponseMock = Array.of(testResultsMockDB[15])
```

Comment:
initialise mocks in beforeEach to allow for parallel tests

Reply:
done